executor/runtime/docker/docker.go

```diff
@@ -375,8 +375,14 @@ func (r *DockerRuntime) computeDNSServers() []string {
 		// the burden on TSA for ipv4 udp traffic
 		return []string{"fd00:ec2::253"}
 	case titus.NetworkConfiguration_Ipv6Only.String():
-		// True ipv6 only means we really can only have a v6
-		// resolver
+		// If we're ipv6-only and running the SystemDNS local resolver
+		// which provides DNS64, we'll point to that
+		if systemServices, err := r.c.SystemServices(); err == nil {
+			if runtimeTypes.GetSidecarConfig(systemServices, runtimeTypes.SidecarServiceSystemDNS).EnabledCheck(&r.cfg, r.c) {
+				return []string{"[::1]:5353"}
+			}
+		}
```

Comment:
so this is a little gnarly but ties the logic here to `shouldStartSystemDNS` rather than duplicating

Reply:
I agree, but is fine

Reply:
Actually, can you do `runtimeTypes.ShouldStartSystemDNS(r.cfg, r.c)` instead?
You will have to capitlize to export, but I think this is fine

Reply:
Ah, was going to do that to start but realized it was unexported in a different package. If there's no reason not to export I'll do it that way 👍